source/horoscope/js/sidebar.js

```diff
@@ -39,6 +39,7 @@ class Horoscope{
 }
 
 let horoscopes = [];
+let horoscopesJSON = new Set();
```

Comment:
Do we need a whole separate variable?

Reply:
We can either have a separate variable, or we can loop through the horoscopes array every time we save a horoscope. Tradeoff for time vs memory. If there is a better way to compare objects in JS, let me know.